test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -84,34 +91,37 @@ public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabl
                 services.AddAzureStorageScaleServices();
 
                 services.AddSingleton<IConcurrencyStatusRepository, TestConcurrencyStatusRepository>();
+                services.AddSingleton<ITriggerMetadataProvider>(new TestTriggerMetadataProvider(triggerMetadata));
             })
             .ConfigureWebJobsScale((context, builder) =>
-                {
-                    builder.UseHostId(hostId);
-                },
-                scaleOptions =>
-                {
-                    scaleOptions.IsTargetScalingEnabled = tbsEnabled;
-                    scaleOptions.MetricsPurgeEnabled = false;
-                    scaleOptions.ScaleMetricsMaxAge = TimeSpan.FromMinutes(4);
-                    scaleOptions.IsRuntimeScalingEnabled = true;
-                });
-
-            foreach (TriggerMetadata t in triggerMetadata)
             {
-                if (t.Type == ATriggerType)
-                {
-                    hostBuilder.ConfigureTestExtensionAScale(t);
-                }
-                else
+                builder.UseHostId(hostId);
+
+                var webJobsBuilderContext = new WebJobsBuilderContext
                 {
-                    hostBuilder.ConfigureTestExtensionBScale(t);
-                }
-            }
+                    Configuration = context.Configuration,
+                    EnvironmentName = context.HostingEnvironment.EnvironmentName
+                };
+                webJobsBuilderContext.Properties["IsAzureWebJobsScaleHost"] = true;
```

Comment:
Still considering the best way to flow this property. An alternative might be to add the "context" HostBuilderContext in this scope as a property on WebJobsBuilderContext. We could then leverage the Property collection of that type, and could also set this flag automatically in ConfigureWebJobsScale.